### DISCUSSION TITLE
Buiding thinks i'm on a different planet

### DISCUSSION THREAD

**Discussion by Soma257** [notnotmelon/factorissimo-2-notnotmelon#137] [General]
Picket up a Building on Nauvis that wasn't completely empty and placed it down on fulgora - i can't make any of the fulgora items in it now cause it thinks i'm still on nauvis:

![aedrthfdrdfenannt](https://github.com/user-attachments/assets/569fce41-61bf-4ea6-bfbe-b6b529ab2b7b)

also we need an option to pickup an empty building so that it becomes a new building (not packed) and it auto deconstructs all items inside into the inventory if there is enough space.

**Comment by SHADOW-XIII** [comment]
looking at the code the factory surface is set when it's first being created
since it's not rocket-transferrable, I imagine it was suppose to stay on the planet it was constructed on

**Reply by Soma257** [reply]
I see, since its easy to transfer them per rocket or other methods with rocket cargo mods or straight up teleporters that wasn't taken into consideration - i assume checking current planet when placing it down again wouldn't be too difficult. Question is what happens with buildings that have recipes that should not be available on the new planet.

Either way having it auto update or a way to update/refresh it would be nice.

**Comment by notnotmelon** [comment]
Changing the planet on-place would involve 
- Serializing all entities placed on that factory building.
- Deleting all entities and tiles in the area.
- Forming a new factory building on the proper planet.
- Huge amounts of mod compatibility checks.

For this reason, such a feature is extremely difficult to implement and outside the scope of this project.
I have added a rocket capacity of 0 in order to resolve this.

Best I can do is check if the factory is the wrong planet and prevent placing it at all.

**Reply by SHADOW-XIII** [reply]
Since "teleportation/gateway" mods exists ... I think a better approach would be to prevent placement if the planet surface (planet) is different to the one of the packed factory.
Some small error like "Cannot be placed here" or "Cannot be placed on this planet"

**Reply by Soma257** [reply]
An error message would at least prevent the confusion of not being able to select certain recipes after placing it on another planet, and maybe adding something to the description of packed buildings like "Can only be placed on x planet" (might be another issue with modded planets tho) would be nice if a refresh button built into the building controller is not an option.
